src/ladds/distributedMemParallelization/AltitudeBasedDecomposition.cpp

```diff
@@ -0,0 +1,176 @@
+#include <spdlog/spdlog.h>
+
+#include "ladds/TypeDefinitions.h"
+#include "satellitePropagator/physics/Constants.h"
+
+LADDS::AltitudeBasedDecomposition::AltitudeBasedDecomposition(LADDS::ConfigReader &config) {
+  auto logger = spdlog::get(LADDS_SPD_LOGGER_NAME);
+  // initialize a one-dimensional MPI world in which split the altitudes
+  int numRanks{};
+  autopas::AutoPas_MPI_Comm_size(AUTOPAS_MPI_COMM_WORLD, &numRanks);
+  // let MPI decide a dimension splitting with no periodicity
+  std::array<int, 1> periods{false};
+  std::array<int, 1> dims{};
+  autopas::AutoPas_MPI_Dims_create(numRanks, dims.size(), dims.data());
+  // create a communicator
+  autopas::AutoPas_MPI_Cart_create(
+      AUTOPAS_MPI_COMM_WORLD, dims.size(), dims.data(), periods.data(), true, &communicator);
+  // get coordinates of this rank within the decomposition grid
+  int rank{};
+  autopas::AutoPas_MPI_Comm_rank(communicator, &rank);
+  std::array<int, 1> coords{};
+  autopas::AutoPas_MPI_Cart_coords(communicator, rank, dims.size(), coords.data());
+
+  // initialize simulation specific data
+  const auto maxAltitude = config.get<double>("sim/maxAltitude");
+  const auto minAltitude = config.get<double>("sim/minAltitude");
+
+  globalBoxMin = {-maxAltitude, -maxAltitude, -maxAltitude};
+  globalBoxMax = {maxAltitude, maxAltitude, maxAltitude};
+
+  // calculate local box extent
+  // we have to account for the fact that there won't be anything between 0, Physics::R_EARTH  + minAltitude, so
+  // shifting the interval a bit
+  // TODO: find a more even distribution of the particles
+  altitudeIntervals = this->logspace(std::log10(minAltitude), std::log10(maxAltitude - Physics::R_EARTH), numRanks);
+  for (int i = 0; i < numRanks; i++) {
+    altitudeIntervals[i] += Physics::R_EARTH;
+  }
+  altitudeIntervals[0] = 0.0;
+  // note that actual max altitude is the corner of the box, so higher than maxaltitude in the config but there should
+  // be very satellites between maxAltitude and this
+  altitudeIntervals.push_back(std::sqrt(3 * std::pow(maxAltitude, 2.0)));
+  SPDLOG_LOGGER_INFO(config.getLogger().get(),
+                     "Computed altitude intervals fo ranks: {}",
+                     autopas::utils::ArrayUtils::to_string(altitudeIntervals));
+
+  localBoxMin = {-altitudeIntervals[rank + 1], -altitudeIntervals[rank + 1], -altitudeIntervals[rank + 1]};
+  localBoxMax = {altitudeIntervals[rank + 1], altitudeIntervals[rank + 1], altitudeIntervals[rank + 1]};
+
+  // print parallelization info
+  if (rank == 0) {
+    SPDLOG_LOGGER_INFO(config.getLogger().get(),
+                       "Parallelization Configuration\n"
+                       "MPI Ranks              : {}\n"
+                       "OpenMP Threads per Rank: {}\n"
+                       "MPI Decomposition      : {}\n",
+                       numRanks,
+                       autopas::autopas_get_max_threads(),
+                       autopas::utils::ArrayUtils::to_string(dims));
+  }
```

Comment:
Hmmm, I am not quite sure how do to that without a lot of code duplication because you need to get rank, numRanks and the communicator again. And if you were to call this constructor too early it wouldn't work because the commworld wouldn't be initialized?

Reply:
1. The point is to reduce code duplication and make sure all Decompositions have the same output-style by using shared functionalities
2. We could also just introduce some utility function that gets us the MPI info because we do this really often.
3. As far as I see it, the parent constructor would only need periods and dims information which can be passed from the child constructor.